src/Microsoft.Health.Dicom.Api/Features/Filters/AcceptContentFilterAttribute.cs

```diff
@@ -46,55 +41,46 @@ public AcceptContentFilterAttribute(string[] mediaTypes, bool allowSingle, bool
                 Debug.Assert(false, "The values in the mediaTypes parameter must be parseable by MediaTypeHeaderValue.");
             }
         }
-
-        _allowSingle = allowSingle;
-        _allowMultiple = allowMultiple;
     }
 
     public override void OnActionExecuting(ActionExecutingContext context)
     {
         EnsureArg.IsNotNull(context, nameof(context));
-        IList<MediaTypeHeaderValue> acceptHeaders = context.HttpContext.Request.GetTypedHeaders().Accept;
 
-        bool acceptable = false;
+        bool acceptable = AcceptHeadersContainKnownTypes(context.HttpContext.Request);
 
-        // Validate the accept headers has one of the specified accepted media types.
-        if (acceptHeaders != null && acceptHeaders.Count > 0)
+        if (!acceptable)
         {
-            foreach (MediaTypeHeaderValue acceptHeader in acceptHeaders)
-            {
-                if (_allowMultiple && StringSegment.Equals(acceptHeader.MediaType, KnownContentTypes.MultipartRelated, StringComparison.InvariantCultureIgnoreCase))
-                {
-                    NameValueHeaderValue typeParameterValue = acceptHeader.Parameters.FirstOrDefault(
-                        parameter => StringSegment.Equals(parameter.Name, TypeParameter, StringComparison.InvariantCultureIgnoreCase));
+            context.Result = new StatusCodeResult(NotAcceptableResponseCode);
+        }
 
-                    if (typeParameterValue != null &&
-                        MediaTypeHeaderValue.TryParse(HeaderUtilities.RemoveQuotes(typeParameterValue.Value), out MediaTypeHeaderValue parsedValue) &&
-                        _mediaTypes.Contains(parsedValue))
-                    {
-                        acceptable = true;
-                        break;
-                    }
-                }
+        base.OnActionExecuting(context);
+    }
 
-                if (_allowSingle)
-                {
-                    string[] split = acceptHeader.ToString().Split(';');
-                    var stringHeaders = _mediaTypes.Select(x => x.ToString()).ToList();
-                    if (split.Any(x => stringHeaders.Contains(x, StringComparer.InvariantCultureIgnoreCase)))
-                    {
-                        acceptable = true;
-                        break;
-                    }
-                }
+    private bool AcceptHeadersContainKnownTypes(HttpRequest request)
+    {
+        IList<MediaTypeHeaderValue> acceptHeaders = ParseAcceptHeaders(request.Headers.Accept);
+
+        foreach (MediaTypeHeaderValue acceptHeader in acceptHeaders)
+        {
+            if (_mediaTypes.Any(x => x.MatchesMediaType(acceptHeader.MediaType)))
```

Comment:
As discussed, we will flip this check - the defined media types for this action should be a subset of the Accept header media range, e.g. `application/dicom` is a subset of `application/*` and so will match.